core/src/main/java/com/amplifyframework/storage/options/StorageGetUrlOptions.java

```diff
@@ -0,0 +1,161 @@
+     * new clone of the provided options, which shares some common
+     * configuration.
+     * @param options Options to populate into a new builder configuration
+     * @return A Builder instance that has been configured using the
+     *         values in the provided options
+     */
+    public static Builder from(StorageGetUrlOptions options) {
+        return builder()
+            .accessLevel(options.getAccessLevel())
+            .targetIdentityId(options.getTargetIdentityId())
+            .expires(options.getExpires());
+    }
+
+    /**
+     * Constructs a default instance of the {@link StorageGetUrlOptions}.
+     * @return default instance of StorageGetUrlOptions
+     */
+    public static StorageGetUrlOptions defaultInstance() {
+        return builder().build();
+    }
+
+    /**
+     * A utility that can be used to configure and construct immutable
+     * instances of the {@link StorageGetUrlOptions}, by chaining
+     * fluent configuration method calls.
+     */
+    public static final class Builder {
+        private StorageAccessLevel accessLevel;
+        private String targetIdentityId;
+        private int expires;
+
+        /**
+         * Configures the storage access level to set on new
+         * StorageGetUrlOptions instances.
+         * @param accessLevel Storage access level for new StorageGetUrlOptions instances
+         * @return Current Builder instance, for fluent method chaining
+         */
+        public Builder accessLevel(StorageAccessLevel accessLevel) {
+            this.accessLevel = accessLevel;
+            return this;
+        }
+
+        /**
+         * Configures the target identity ID that will be used on newly
+         * built StorageGetUrlOptions.
+         * @param targetIdentityId Target identity ID for new StorageGetUrlOptions instances
+         * @return Current Builder instance, for fluent method chaining
+         */
+        public Builder targetIdentityId(String targetIdentityId) {
+            this.targetIdentityId = targetIdentityId;
+            return this;
+        }
+
+        /**
+         * Configures the number of seconds left until URL expires on new
+         * StorageGetUrlOptions instances.
+         * StorageGetUrlOptions instances.
+         * @param accessLevel Storage access level for new StorageGetUrlOptions instances
+         * @return Current Builder instance, for fluent method chaining
+         */
+        public Builder expires(int accessLevel) {
+            this.expires = expires;
+            return this;
+        }
+
+        /**
+         * Constructs and returns a new immutable instance of the
+         * StorageGetUrlOptions, using the configurations that
+         * have been provided the current instance of the Builder.
+         * @return A new immutable instance of StorageGetUrlOptions
+         */
+        public StorageGetUrlOptions build() {
+            return new StorageGetUrlOptions(this);
```

Comment:
Do you want to validate anything about the provided values before you build the instance? I.e. a range on the int, null-ness is the provided objects?

Reply:
To be revisited in `StorageOptions` PR